pennylane/ops/qubit/state_preparation.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the discrete-variable quantum operations concerned
+with preparing a certain state on the device.
+"""
+# pylint:disable=abstract-method,arguments-differ,protected-access
+
+from pennylane.operation import AnyWires, Operation
+from pennylane.templates.state_preparations import BasisStatePreparation, MottonenStatePreparation
+
+
+class AdjointError(Exception):
```

Comment:
This would seem to be more general, though only used for state preps atm. Maybe worth moving it into the `__init__.py` or some other more general place?

Reply:
I think having it in `__init__.py` sounds like a good idea.

Reply:
Perhaps addressing this as well, moving the exception to the `__init__.py` file would make sense here, and simply import it from there.